Foundation/URLCache.swift

```diff
@@ -193,7 +326,123 @@ open class URLCache : NSObject {
         @result an initialized URLCache, with the given capacity, backed
         by disk.
     */
-    public init(memoryCapacity: Int, diskCapacity: Int, diskPath path: String?) { NSUnimplemented() }
+    public init(memoryCapacity: Int, diskCapacity: Int, diskPath path: String?) {
+        self.memoryCapacity = memoryCapacity
+        self.diskCapacity = diskCapacity
+        
+        if let path = path {
+            cacheDirectory = URL(fileURLWithPath: path)
+        } else {
+            do {
+                let caches = try FileManager.default.url(for: .cachesDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+                let directoryName = (Bundle.main.bundleIdentifier ?? ProcessInfo.processInfo.processName)
+                    .replacingOccurrences(of: "/", with: "_")
+                    .replacingOccurrences(of: "\\", with: "_")
+                    .replacingOccurrences(of: ":", with: "_")
+
+                // We append a Swift Foundation identifier to avoid clobbering a Darwin cache that may exist at the same path;
+                // the two on-disk cache formats aren't compatible.
+                let url = caches
+                    .appendingPathComponent("org.swift.Foundation.URLCache", isDirectory: true)
+                    .appendingPathComponent(directoryName, isDirectory: true)
+                try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
```

Comment:
@drodriguez IIRC on Darwin the path, if passed, must exist, hence the logic here. I'll double-check that this is the case.

Reply:
It was indeed not the case.

Reply:
Fixed.